### ISSUE TITLE
Warning when using container_run_and_commit

### ISSUE THREAD

**Issue by joprice** [bazelbuild/rules_docker#1327]
I am trying to use `container_run_and_commit`, `container_image` and `container_push` to build a docker image. The `container_run_and_commit` rule is need to install some extra dependencies on top of the base image. When I build image, I get the following error:

```Pushing an image based on a tarball can be very expensive.  If the image is the output of a container_build, consider dropping the '.tar' extension. If the image is checked in, consider using container_import instead.```

This references a rule `container_build`, which doesn't seem to exist anymore. I'm not sure if I'm doing something wrong or if this is a spurious warning. The suggestion to remove ".tar" is also not applicable since I'm simply referencing the previous rule each time, not some specific target with tar in it:

```python
container_run_and_commit(
  name = "install_plugins",
  image = "@some-base-image",
  commands = [...],
  ...
)

container_image(
  name = "image",
  base = ":install_plugins",
  ...
)

container_push(
  image = ":image",
  ...
)
```

Is there another recommended way to use the rules for a use case like this?

**Comment by nlopezgi** [comment]
What you are seeing is not an error, its a warning. Unfortunately using `container_run_and_commit` requires this expensive operation, iirc. Please let me know if you are actually experiencing this as an error somehow. Closing for now, but let me know to reopen for some reason.

**Comment by joprice** [comment]
My issue was with the content of the message. It seems to be incorrect since it mentions a non-existent rule.

**Comment by nlopezgi** [comment]
Yeah, you're right

**Comment by joprice** [comment]
I could submit a pr to update the comment but I'm not sure what it should recommend. I'm poking around in the source to see if I can find a better way to do this. I'd rather not rely on a something that causes a warning. Is it generally not recommended to use these rules for docker files that require a step that runs arbitrary bash?

**Comment by lazcamus** [comment]
Found this via Google search.

Disappointing that using `container_run_and_commit` spews an error message admonishing me for doing something expensive every time. I guess the intention is good (?) but the message is misleading.
